apps/web/app/components/FileUploadInput/FileUploadInput.tsx

```diff
@@ -0,0 +1,121 @@
+import DocViewer, { DocViewerRenderers } from "@cyntler/react-doc-viewer";
+import { useEffect, useState } from "react";
+
+import { cn } from "~/lib/utils";
+import { ContentTypes } from "~/modules/Admin/EditCourse/EditCourse.types";
+
+import { Icon } from "../Icon";
+
+import type React from "react";
+
+type FileUploadProps = {
+  handleFileUpload: (file: File) => Promise<void>;
+  isUploading: boolean;
+  contentTypeToDisplay: string;
+  url?: string;
+};
+
+const FileUploadInput = ({
+  handleFileUpload,
+  isUploading,
+  contentTypeToDisplay,
+  url,
+}: FileUploadProps) => {
+  const [videoPreview, setVideoPreview] = useState<string | null>(null);
+  const [file, setFile] = useState<File | null>(null);
+  const acceptedTypes =
+    contentTypeToDisplay === ContentTypes.VIDEO_LESSON_FORM ? ".mp4,.avi,.mov" : ".pptx,.ppt,.odp";
+
+  useEffect(() => {
+    if (url) {
+      setVideoPreview(url);
+    }
+  }, [url]);
+
+  const handleFileChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
+    const uploadedFile = e.target.files?.[0];
+    if (uploadedFile) {
+      setFile(uploadedFile);
+      if (contentTypeToDisplay === ContentTypes.VIDEO_LESSON_FORM) {
+        const videoURL = URL.createObjectURL(uploadedFile);
+        setVideoPreview(videoURL);
+      }
+      await handleFileUpload(uploadedFile);
+    }
+  };
+
+  const docs = url
+    ? [
+        {
+          uri: url as string,
+          fileType: "pptx",
+          fileName: "Presentation",
+        },
+      ]
+    : [];
+
+  return (
+    <div className="flex  flex-col gap-y-2">
+      <div className="w-3/5 h-[200px] md:h-[350px] xl:h-[500px] rounded-lg border-solid border-2 border-gray-300 flex bg-gray-100 relative cursor-pointer overflow-hidden flex-col">
+        {!videoPreview && !file ? (
```

Comment:
What do you think about extracting it to the component just to get rid of this nested ternary? Imo this makes it difficult to read the code